src/Week2/LogAnalysis.hs

```diff
@@ -0,0 +1,104 @@
+{-# OPTIONS_GHC -Wall #-}
+module Week2.LogAnalysis where
+
+import Week2.Log
+
+{-
+Exercise 1
+
+Define a function
+parseMessage :: String -> LogMessage
+which parses an individual line from the log file. 
+
+For example,
+ parseMessage "E 2 562 help help"
+     == LogMessage (Error 2) 562 "help help"
+parseMessage "I 29 la la la"
+     == LogMessage Info 29 "la la la"
+ parseMessage "This is not in the right format"
+     == Unknown "This is not in the right format"
+
+Once we can parse one log message, we can parse a whole log file. 
+Define a function parse :: String -> [LogMessage]
+which parses an entire log file at once and returns its contents as a list of LogMessages.
+-}
+
+parseMessage :: String -> LogMessage
+parseMessage s =
+    let words_list = words s in
+    case words_list of
+    ("I":num:message) -> LogMessage Info (read num) (unwords message)
+    ("W":num:message) -> LogMessage Warning (read num) (unwords message)
+    ("E":val:num:message) -> LogMessage (Error (read val)) (read num) (unwords message)
+    _ -> Unknown (unwords words_list)
```

Comment:
Minor style nitpicks:
- Follow a 4 space indentation pattern. That's the convention we follow for CDMP. 
- Add one space on either side of `:`. Like `("W" : num : message)`. Greatly improves readability.